src/transformStateWithClones.js

```diff
@@ -7,7 +7,36 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateCopy = { ...state };
+  const changesArray = [];
+
+  for (const deed of actions) {
+    switch (deed.type) {
+      case 'addProperties':
+        for (const key in deed.extraData) {
+          stateCopy[key] = deed.extraData[key];
+        };
+        break;
+
+      case 'removeProperties':
+        for (let j = 0; j < deed.keysToRemove.length; j++) {
+          delete stateCopy[deed.keysToRemove[j]];
+        };
+        break;
+
+      case 'clear':
+        for (const lock in stateCopy) {
```

Comment:
```suggestion
        for (const key in stateCopy) {
```
You can use the same names